object-mapper.rst

```diff
@@ -0,0 +1,724 @@
+Object Mapper
+=============
+
+Symfony provides a mapper component to transform a given object into another one,
+simplifying tasks like converting DTOs (Data Transfer Objects) to entities or
+vice-versa.
```

Comment:
We could add "This component is especially useful when decoupling API input/output from internal models."